src/elements/content-uploader/ItemAction.tsx

```diff
@@ -61,41 +38,51 @@ const ItemAction = ({
     onUpgradeCTAClick,
     status,
 }: ItemActionProps) => {
-    const intl = useIntl();
-    let icon: React.ReactNode = <XMark color="black" height={Size5} width={Size5} />;
-    let tooltip;
-    let isLoading = false;
+    const { formatMessage } = useIntl();
     const { code } = error || {};
-    const { formatMessage } = intl;
+
+    let Icon = XMark;
+    let tooltip;
 
     if (isFolder && status !== STATUS_PENDING) {
         return null;
     }
 
     switch (status) {
         case STATUS_COMPLETE:
-            icon = <Checkmark aria-label="complete" color={SurfaceStatusSurfaceSuccess} height={Size5} width={Size5} />;
+            Icon = () => (
+                <Checkmark
+                    aria-label={formatMessage(messages.complete)}
+                    color={SurfaceStatusSurfaceSuccess}
+                    height={Size5}
+                    width={Size5}
+                />
+            );
             if (!isResumableUploadsEnabled) {
                 tooltip = messages.remove;
             }
             break;
         case STATUS_ERROR:
-            icon = <ArrowCurveForward aria-label="error" color="black" height={Size5} width={Size5} />;
+            Icon = ArrowCurveForward;
             tooltip = isResumableUploadsEnabled ? messages.resume : messages.retry;
             break;
         case STATUS_IN_PROGRESS:
         case STATUS_STAGED:
             if (isResumableUploadsEnabled) {
-                isLoading = true;
+                Icon = () => (
+                    <LoadingIndicator aria-label={formatMessage(messages.loading)} className="bcu-ItemAction-loading" />
+                );
             } else {
-                icon = <EllipsisBadge aria-label="staged" color="black" height={Size5} width={Size5} />;
+                Icon = IconInProgress;
                 tooltip = messages.uploadsCancelButtonTooltip;
             }
             break;
         case STATUS_PENDING:
         default:
             if (isResumableUploadsEnabled) {
-                isLoading = true;
+                Icon = () => (
+                    <LoadingIndicator aria-label={formatMessage(messages.loading)} className="bcu-ItemAction-loading" />
```

Comment:
I saw this line is duplicate with the STATUS_STAGED case above. Could we deduplicate it by having it written separately since the LoadingIndicator would kind be fixed.

`const loadingIndicatorIcon = () => <LoadingIndicator aria-label={formatMessage(messages.loading)} className="bcu-ItemAction-loading" />`